internal/pkg/agent/install/install.go

```diff
@@ -53,15 +58,36 @@ func Install(cfgFile string) error {
 
 	// place shell wrapper, if present on platform
 	if paths.ShellWrapperPath != "" {
-		err = os.MkdirAll(filepath.Dir(paths.ShellWrapperPath), 0755)
-		if err == nil {
-			err = ioutil.WriteFile(paths.ShellWrapperPath, []byte(paths.ShellWrapper), 0755) // nolint:gosec,G306 // this is fine.
-		}
-		if err != nil {
-			return errors.New(
-				err,
-				fmt.Sprintf("failed to write shell wrapper (%s)", paths.ShellWrapperPath),
-				errors.M("destination", paths.ShellWrapperPath))
+		// Install symlink for darwin instead
```

Comment:
Can we explain why we need to do this? Let's document the reason we need this symlink in the code or in the `templates/darwin/elastic-agent.tmpl` script itself for future engineers to understand why this was necessary.

Reply:
Before this change the agent used the shell script wrapper to start the agent, this posed an issue with FDA cause the root process of the launchd service was `sh` instead of the elastic-agent, so in order to grant FDA the FDA needed to be granted to ```/bin/sh```. Here we replace that wrapper shell script with symlink, so the root process of the service is the ```elastic-agent```. Will add more comments here for main branch backports.